engine/common/mod_bmodel.c

```diff
@@ -1881,313 +2029,352 @@ static void Mod_LoadMarkSurfaces( dbspmodel_t *bmod )
 	}
 }
 
-/*
-=================
-Mod_LoadTextures
-=================
-*/
-static void Mod_LoadTextures( dbspmodel_t *bmod )
-{
-	dmiptexlump_t	*in;
-	texture_t		*tx, *tx2;
-	texture_t		*anims[10];
-	texture_t		*altanims[10];
-	int		num, max, altmax;
-	qboolean		custom_palette;
-	char		texname[64];
-	mip_t		*mt;
-	int 		i, j;
+static void Mod_LoadTextureData( dbspmodel_t* bmod, int textureIndex )
+{
+#if !XASH_DEDICATED
+	texture_t* texture = NULL;
```

Comment:
`texture_t *texture;`
`mip_t *mipTex;`

No reason to initialize them to NULL, as you initialize them at the start of the function anyway.